buffer_pool_for_uploading.go

```diff
@@ -1,27 +1,46 @@
 package goseaweedfs
 
-import "github.com/oxtoacart/bpool"
+import (
+	"bytes"
+	"sync"
+)
 
-// !! For user who wants to use buffer pool when uploading, please explicitly call SetBufferPoolForUploading(poolSize, BufferSize) !!
+var gBufferPool *BufferPool
 
-var gBufferPool *bpool.SizedBufferPool
-
-////BufferPoolOptionsForUploading : will be used when uploading with multipart which will pre-allocate and reuse memory, and reduce memory usage significantly if we can estimate the file size we are uploading.
-//type BufferPoolOptionsForUploading struct {
-//	PoolSize   int // how many buffers in the pool, eg: 16
-//	BufferSize int // length of a buffer, eg: 32 * 1024 * 1024
-//}
-//
-//func NewBufferPoolOptionsForUploading(poolSize int, bufferSize int) *BufferPoolOptionsForUploading {
-//	return &BufferPoolOptionsForUploading{PoolSize: poolSize, BufferSize: bufferSize}
-//}
-
-//SetUploadOptions : will be used when uploading with multipart which will pre-allocate and reuse memory, and reduce memory usage significantly if we can estimate the size of uploading file.
-func SetBufferPoolForUploading(poolSize, bufferSize int) *bpool.SizedBufferPool {
-	gBufferPool = bpool.NewSizedBufferPool(poolSize, bufferSize)
+// !! For user who wants to use buffer pool when uploading, please explicitly call InitBufferPool(bufferLen, bufferCap int) !!
+func InitBufferPool(bufferLen, bufferCap int) *BufferPool {
+	gBufferPool = NewBufferPool(bufferLen, bufferCap)
+	buf := gBufferPool.Get() // pre-cache
+	gBufferPool.Put(buf)     // pre-cache
 	return gBufferPool
 }
 
-func GetBufferPoolForUploading() *bpool.SizedBufferPool {
+func GetBufferPool() *BufferPool {
 	return gBufferPool
 }
+
+// BufferPool : will be used when uploading with multipart which will pre-allocate and reuse memory, and reduce memory usage significantly if we can estimate the file size we are uploading.
+type BufferPool struct {
+	BufferLen int
+	BufferCap int
```

Comment:
Buffer should care about `capacity` only. 

Everytime buffer is fetched for writing, `bytes.Buffer` must reset, thus make its len become 0. Therefore, care about `BufferLen` is redundant.

Reply:
OK.

Reply:
go.mod and go.sum is annoying while forking and merging, so it might be trash in the commit